suitcase/tiff_series/__init__.py

```diff
@@ -264,38 +249,30 @@ def event_page(self, doc):
             EventPage document
         '''
         event_model.verify_filled(doc)
-        streamname = self._streamnames[doc['descriptor']]
+        descriptor = self._descriptors[doc['descriptor']]
+        streamname = descriptor.get('name')
         for field in doc['data']:
             for img in doc['data'][field]:
                 # check that the data is 2D, if not ignore it
                 img_asarray = numpy.asarray(img)
                 if img_asarray.ndim == 2:
-                    if self._stack_images:
-                        # create a file for this stream and field if required
-                        if not self._tiff_writers.get(streamname, {}).get(field):
-                            filename = (f'{self._templated_file_prefix}'
-                                        f'{streamname}-{field}.tiff')
-                            file = self._manager.open(
-                                'stream_data', filename, 'xb')
-                            tw = TiffWriter(file, **self._init_kwargs)
-                            self._tiff_writers[streamname][field] = tw
-                        # append the image to the file
-                        tw = self._tiff_writers[streamname][field]
-                        tw.save(img_asarray, *self._kwargs)
+                    # template the file name.
+                    self._templated_file_prefix = self._file_prefix.format(
+                        start=self._start, descriptor=descriptor,
+                        event_page=doc)
+                    if not (self._counter.get(streamname, {}).get(field) or
```

Comment:
0 is also falsy so we don't need the second part of this `or`

Reply:
done